python/ray/util/spark/start_ray_node.py

```diff
@@ -109,8 +110,32 @@ def try_clean_temp_dir_at_exit():
             fcntl.flock(lock_fd, fcntl.LOCK_UN)
             os.close(lock_fd)
 
+    def check_parent_alive() -> None:
+        orig_parent_id = os.getppid()
+        while True:
+            time.sleep(0.5)
+            if os.getppid() != orig_parent_id:
+                process.terminate()
+                try_clean_temp_dir_at_exit()
+                os._exit(143)
+
+    threading.Thread(target=check_parent_alive, daemon=True).start()
+
```

Comment:
You already have a separate PR for it.

Reply:
Yes. We should merge https://github.com/ray-project/ray/pull/38652 firstly, then merge this PR.

Reply:
Merged.

Reply:
This now can be removed.